src/file-selector.ts

```diff
@@ -121,6 +121,15 @@ function flatten<T>(items: any[]): T[] {
 }
 
 function fromDataTransferItem(item: DataTransferItem) {
+    if (typeof (item as any).getAsFileSystemHandle === 'function') {
+        return (item as any).getAsFileSystemHandle()
+            .then(async (h: any) => {
+                const file = await h.getFile();
+                file.handle = h;
+                return toFileWithPath(file);
+            })
+            .catch((error: {message: any}) => error.message);
```

Comment:
```suggestion
            .catch((error) => error);
```

Reply:
What I meant is to let the promise fail so that the error can be caught by the user instead of catching it and returning the error (which is not a file and would like cause issues).